### DISCUSSION TITLE
Security issue in AP-mode

### DISCUSSION THREAD

**Discussion by scanheads** [arendst/Tasmota#18845] [Q&A]
As the link for security-related problems only leads to nowhere (got a "404 Object not found!
The requested URL was not found on this server. If you entered the URL manually please check your spelling and try again."), I report this here. There is a serious problem in Tasmota's AP-mode which makes it easy as fuck to take over an existing Tasmota device. This can be done quite easy:

1. Jam all existing WiFi networks close to the Tasmota-device to be attacked (this is extremely simple as there are ready-to-use applications available e.g. for the ESP32)
2. Now Tasmota comes up in AP-mode
3. Exclude the Tasmota-AP from the WiFi networks to be jammed
4. Connect to the Tasmota-device and do whatever evil you want to do (turn it on/off, change configuration, damage configuration, change stored values, configure it to connect to an other WiFi network so take over control permanently, set some Web-passwords,...)

The problem with Tasmota is, there is no possibility to set some password for the AP-mode. Whenever the AP-mode is enabled, it comes as open network without any protection and gives full access to everything.

So the solution is also very simple: give the possibility to configure a password for the default/fallback AP so that it no longer comes up as open network.

Yes, I'm aware of the web admin password, but this is not a solution but a pain in the a**. It turns on a password protection for normal operation, so whenever one wants to access the web interface regularly, one has to enter this password. Or in other words: even people that are willing to configure Tasmota in a secure way will not make use of this "feature" as it disturbs them way too much during regular operation.

**Comment by barbudor** [comment]
> The problem with Tasmota is, there is no possibility to set some password for the AP-mode

Wrong
```
#define WIFI_AP_PASSPHRASE     ""                // AccessPoint passphrase. For WPA2 min 8 char, for open use "" (max 63 char).
```
Documented in the changelog


> As the link for security-related problems only leads to nowhere (got a "404 Object not found

If you found a problem in the docs, please open an issue in the doc repository at https://github.com/tasmota/docs

Better, you can also contribute to the doc to improve it by clicking on the pencil button on the top right of the doc page.

Thanks in advance

**Reply by scanheads** [reply]
> #define WIFI_AP_PASSPHRASE     ""                // AccessPoint passphrase. For WPA2 min 8 char, for open use "" (max 63 char).

No, this is not a solution. This works for the few people that know of things like GitHub or a compiler, but the millions of people, that buy a ready-to-use Tasmota device can ONLY do what is possible in the web interface. So as long as this password can't be configured easily and via the normal UI, this security issue will exist.

**Reply by sfromis** [reply]
No, it will not. See what I posted.

**Reply by barbudor** [reply]
If we were applying a standard passphrase, it would be documented so no more secure than the absence of passphrase.
Providing a command to set a passphrase would be as fragile as settings can be erased for safe recovery which would revert to the same.

Tasmota project do not provide hardware but just a software that needs to be configured.
There is a way to do it.
It 's up to users or manufacturers that are using Tasmota for free to take their responsibilities. 

All my devices are running a custom built firmware where AP passphrase is customized and WifiConfig 4 is hardcoded as default.

**Reply by sfromis** [reply]
With `WifiConfig 4`, an intruder would need to have the Wifi security already penetrated, or physical access, to get into Tasmota devices. No worse security to this type of attack, compared to other Wifi devices.

**Reply by scanheads** [reply]
@barbudor nobody is talking about a standard passphrase as this would make any sense (for exactly the reasons you described). But what is required, to make the Tasmota devices secure, that are sold with a preinstalled software, would be a passphrase that can be set in the WiFi configuration freely - so after setting the device into operation initially, the user can define its very own AP-password there.

**Reply by sfromis** [reply]
Just stop running with `Wificonfig 2`, to instead follow the repeated recommendation of using `Wificonfig 4`. Then there is no security issue with knocking the intended AP offline.

**Comment by sfromis** [comment]
Additionally, with the intended AP unavailable, Tasmota does not come up in AP mode, if using the recommended (default) `WifiConfig 4` (or 5) - which you really should if wanting rudimentary security.

https://tasmota.github.io/docs/Securing-your-IoT-from-hacking/#disable-unsecured-fallback-wifi-wifimanager

**Reply by scanheads** [reply]
Is this WifiConfig-mode enabled by default? No. So this does not help as such hacks apart from the GUI are not obvious for "normal" users - or in other words: most users are not aware of this and therefore are not able to set this configuration.

**Reply by sfromis** [reply]
Tasmota is very firmly not aimed as a commercial product for what you call "normal users", but for DIY people willing to work with what they have flashed. If some vendor sells a commercial product with Tasmota badly configured with `Wificonfig 2`, that error should be fixed.

**Reply by sfromis** [reply]
As a test (just to make sure), I did a default installation of an ESP8266, without doing **any** config other than connecting to Wifi. Result is the expected default, which I already stated:
`{"WifiConfig":{"4":"Retry"}}`
This was both after onboarding from the Tasmota Web Installer, or via connecting to the AP broadcast when no network has been configured.

Thus, your "security issue" does not apply.

**Reply by sfromis** [reply]
Of course, someone could have thought "it is more user-friendly with `WifiConfig 2`, then the user can just reconnect if moving the device to the range of another AP", without even thinking about security.
